oprf/server.go

```diff
@@ -0,0 +1,144 @@
+package oprf
+
+import (
+	"crypto/rand"
+	"crypto/subtle"
+	"errors"
+)
+
+// Server is a representation of a Server during protocol execution.
+type Server struct {
+	suite
+	privateKey PrivateKey
+}
+
+// NewServerWithKey creates a new instantiation of a Server. It can create
+// a server with existing keys or use pre-generated keys.
+func NewServerWithKey(id SuiteID, m Mode, skS *PrivateKey) (*Server, error) {
+	// Verifies keypair corresponds to SuiteID.
+	if id != skS.SuiteID {
+		return nil, errors.New("keys don't match with suite")
+	}
+
+	suite, err := suiteFromID(id, m)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Server{*suite, *skS}, nil
+}
+
+// NewServer creates a new instantiation of a Server.
+func NewServer(id SuiteID, m Mode) (*Server, error) {
```

Comment:
Likewise for the client, should we split this up into two modes?

```suggestion
func NewServer(id SuiteID) (*Server, error) {
   // base mode
}
func NewVerifiableServer(id SuiteID, skS *PrivateKey) (*Server, error) {
   // verifiable mode
}   
```